File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    num=0
    n=lottos.count(0)
    for i in range(n):lottos.remove(0)
    for i in range(len(lottos)):num+=win_nums.count(lottos[i])
    if num<2: 
        answer.append(6)
        answer.insert(0,answer[0]-n)
    elif num==2: 
        answer.append(5)
        answer.insert(0,answer[0]-n)
    elif num==3: 
        answer.append(4)
        answer.insert(0,answer[0]-n)
    elif num==4: 
        answer.append(3)
        answer.insert(0,answer[0]-n)
    elif num==5: 
        answer.append(2)
        answer.insert(0,answer[0]-n)
    elif num==6: 
        answer.append(1)
        answer.insert(0,answer[0]-n)
    if answer[0]<1 : answer[0]=1
    return answer

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.kt
class Solution {
    fun solution(lottos: IntArray, win_nums: IntArray): IntArray {
        var answer: IntArray = intArrayOf()
        var zero = lottos.count{it==0}
        var count = 0
        for(i in 0 until lottos.size){
            count += lottos.count{it==win_nums[i]}
        }
        var max = 0
        var min = 0
        when(count){
            6 -> min=1
            5 -> min=2
            4 -> min=3
            3 -> min=4
            2 -> min=5
            else -> min=6
        }
        max=min-zero
        if(max<1) max=1
        answer = answer.plus(max)
        answer = answer.plus(min)
        return answer
    }
} //최저 순위 = 그냥 0인 상태에서 일치하는 숫자 갯수만
  //최고 순위 = 0 두개가 일치한다고 하면 최고 순위